src/servers/src/http/prometheus.rs

```diff
@@ -719,22 +719,63 @@ pub async fn label_values_query(
     }
 
     let queries = params.matches.0;
-    if queries.is_empty() {
-        return PrometheusJsonResponse::error(
-            StatusCode::InvalidArguments,
-            "match[] parameter is required",
-        );
-    }
-
     let start = params.start.unwrap_or_else(yesterday_rfc3339);
     let end = params.end.unwrap_or_else(current_time_rfc3339);
     let lookback = params
         .lookback
         .unwrap_or_else(|| DEFAULT_LOOKBACK_STRING.to_string());
 
-    let mut label_values = HashSet::new();
+    // If no match[] is specified, collect label values from all tables.
+    if queries.is_empty() {
+        let mut label_values = HashSet::new();
+        let mut merge_map = HashMap::new();
+        let table_names = match handler
+            .catalog_manager()
+            .table_names(&catalog, &schema, Some(&query_ctx))
+            .await
+        {
+            Ok(tables) => tables,
+            Err(e) => return PrometheusJsonResponse::error(e.status_code(), e.output_msg()),
+        };
 
+        for table_name in table_names {
+            let prom_query = PromQuery {
+                query: table_name,
+                start: start.clone(),
+                end: end.clone(),
+                step: DEFAULT_LOOKBACK_STRING.to_string(),
+                lookback: lookback.clone(),
+            };
+            let result = handler.do_query(&prom_query, query_ctx.clone()).await;
```

Comment:
Executing queries in a loop may introduce huge latency. We can run these tasks in parallel.

But I still think we need to add some restrictions because there can be a large number of tables to query.

Reply:
agreed seems it need to discuss, and I will make this Pull request to draft